src/main/java/org/springframework/data/mapping/PersistentProperty.java

```diff
@@ -276,6 +281,24 @@ default Field getRequiredField() {
 	 */
 	boolean isAssociation();
 
+	/**
+	 * @return {@literal true} if the property should be embedded.
+	 * @since 2.5
+	 */
+	default boolean isEmbedded() {
+		return isEntity() && findAnnotation(Embedded.class) != null;
+	}
+
+	/**
+	 * @return {@literal true} if the property generally allows {@literal null} values;
+	 * @since 2.5
+	 */
+	default boolean isNullable() {
+
+		return (isEmbedded() && findAnnotation(Embedded.class).onEmpty().equals(OnEmpty.USE_NULL))
```

Comment:
When a property is not embedded, it is never nullable. If that is intended, it should be described in the documentation and the superfluous parens around the first `&&` term should be removed. 

Or the expression should get fixed. Also we should have tests for the not embedded case.